app.rb

```diff
@@ -0,0 +1,84 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/user'
+require './lib/post'
+require './lib/comment'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    @user = User.find_by_id(session[:user_id])
```

Comment:
What happens if `session[:user_id]` is `nil`? Even if this in theory *should not* happen on this page, often unusual cases will come up and can break the application. There should probably be a guard condition handling this, and maybe redirecting to a login page

Reply:
Also, since we're getting a list of posts - should the variable be named `@posts`?

Reply:
> What happens if `session[:user_id]` is `nil`? Even if this in theory _should not_ happen on this page, often unusual cases will come up and can break the application. There should probably be a guard condition handling this, and maybe redirecting to a login page

I have an if else statement in the homepage therefore if it is nil, the page will only display the posts and no ability to add posts or comments with a login button and sign up button that take you to a login page, therefore, it shouldn't be an issue. Or am I missing something?